src/shogun/lib/observers/ParameterObserver.cpp

```diff
@@ -82,3 +82,17 @@ index_t ParameterObserver::get_num_observations() const
 {
 	return utils::safe_convert<index_t>(m_observations.size());
 };
+
+std::shared_ptr<ObservedValue> ParameterObserver::execute_lambda(const TimedObservedValue &value)
+{
+    try
+    {
+        auto func =
+            value.first->get_any().as<std::function<std::shared_ptr<ObservedValue>()>>();
```

Comment:
@gf712 I have a question here. Let's say my `Any` object contains an `ObservedValueTemplated` pointer object. The class `ObservedValueTemplated` inherits from `ObservedValue`. Ideally, I should be able to cast my `Any` to an `ObservedValue` pointer object right (e.g., `any.as<std::shared_ptr<ObservedValue>>()`)?

I thought it would have been possible to do so thanks to the inheritance, but it does not seem to work :/ is this an intended behaviour (or am I missing something)?

Reply:
Yea, it doesn't work off the bat. You need to register that inheritance using
https://github.com/shogun-toolbox/shogun/blob/7014676a95ec27b51473f037c4c137083075d61e/src/shogun/lib/any.h#L1259-L1275

Reply:
This is basically a vtable, but it is only populated with SGObject derived classes to SGObject and some primitive types + std::string. I guess you could always register it there? @vigsterkr